framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -748,118 +789,171 @@ def set_monitor_equation(self, **data):
+                                                          last_form_url=self.last_loaded_form_url)
         return rendering_rules.to_dict()
 
     @cherrypy.expose
+    @using_template("simulator_fragment")
     @handle_error(redirect=False)
     @check_user
-    def launch_pse(self, **data):
-        # TODO: Split into: set range values and Launch, show message with finished config and nr of simulations
+    def set_pse_range_params(self, **data):
+        is_simulator_copy = common.get_from_session(common.KEY_IS_SIMULATOR_COPY) or False
+        is_simulator_load = common.get_from_session(common.KEY_IS_SIMULATOR_LOAD) or False
         all_range_parameters = self.range_parameters.get_all_range_parameters()
-        range_param1, range_param2 = SimulatorPSEParamRangeFragment.fill_from_post(all_range_parameters, **data)
+
+        if cherrypy.request.method == 'POST':
+            is_simulator_copy = False
+            self._update_last_loaded_fragment_url(SimulatorWizzardURLs.LAUNCH_PSE_URL)
+            range_param1, range_param2 = SimulatorPSEParamRangeFragment.fill_from_post(all_range_parameters, **data)
+            common.add2session(common.KEY_PSE_PARAM_1, range_param1)
+            common.add2session(common.KEY_PSE_PARAM_2, range_param2)
+        else:
+            range_param1, range_param2 = self._handle_range_params_at_loading()
+
+        number_of_simulations = len(range_param1.get_range_values())
+
+        if range_param2:
+            number_of_simulations = number_of_simulations * len(range_param2.get_range_values())
+
+        next_form = SimulatorPSEFinalFragment()
+        next_form.fill_from_post({'number_of_simulations': str(number_of_simulations)})
+
+        rendering_rules = SimulatorFragmentRenderingRules(next_form, SimulatorWizzardURLs.LAUNCH_PSE_URL,
+                                                          SimulatorWizzardURLs.SET_PSE_RANGE_PARAMS_URL,
+                                                          is_simulator_copy, is_simulator_load,
+                                                          last_form_url=self.last_loaded_form_url,
+                                                          is_launch_pse_fragment=True)
+        return rendering_rules.to_dict()
+
+    @cherrypy.expose
+    @handle_error(redirect=False)
+    @check_user
+    def launch_pse(self, **data):
         session_stored_simulator = common.get_from_session(common.KEY_SIMULATOR_CONFIG)
+        session_stored_simulator.simulation_length = float(common.get_from_session(common.KEY_SIMULATION_LENGTH))
 
         project = common.get_current_project()
         user = common.get_logged_user()
 
         burst_config = common.get_from_session(common.KEY_BURST_CONFIG)
         burst_config.start_time = datetime.now()
-        # if burst_name != 'none_undefined':
-        #     burst_config.name = burst_name
+        simulation_name = common.get_from_session(common.KEY_SIMULATION_NAME)
```

Comment:
Why this is kept separately in session, why not set it already on the object `burst_config ` in session?

Reply:
Same reason as for the question about the simulation_length setting on line 935.